pkg/cli/secretsencrypt/secrets_encrypt.go

```diff
@@ -0,0 +1,225 @@
+package secretsencrypt
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/tabwriter"
+
+	"github.com/erikdubbelboer/gspt"
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	"github.com/rancher/k3s/pkg/clientaccess"
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/server"
+	"github.com/rancher/k3s/pkg/version"
+	"github.com/urfave/cli"
+	"k8s.io/utils/pointer"
+)
+
+func commandPrep(app *cli.Context, cfg *cmds.Server) (config.Control, *clientaccess.Info, error) {
+	var controlConfig config.Control
+	var err error
+	// hide process arguments from ps output, since they may contain
+	// database credentials or other secrets.
+	gspt.SetProcTitle(os.Args[0] + " encrypt")
+
+	nodeName := app.String("node-name")
+	if nodeName == "" {
```

Comment:
Does this take into consideration the `--with-node-id` flag?
https://github.com/k3s-io/k3s/blob/8271d98a766b060463bc73ef66c5085b5797b4cc/pkg/agent/config/config.go#L355-L361

If it turns out we actually don't need the node name for anything, you might just be able to drop this bit.

Reply:
Removed this, tested with different node-name and with node id flags... didn't affect rotation.